frontend/components/sign-up/js/sign-up.component.ts

```diff
@@ -185,6 +185,8 @@ class SignupController implements ng.IController {
 		this.$scope.$watch(() =>  this.$scope.signup.$error, (error) => {
 			if (error.email) {
 				this.emailInvalid = true;
+			} else if (!error.email) {
```

Comment:
Instead of storing a global variable, can we just access  this on doRegister? (the only place that we check this)

I feel like a continuous watch variable just to flag a boolean that is only used in one place sounds a bit overkill. But then again I'm not really sure how these scope fields work.

Reply:
Yep, that's a good call. We can just access it when we do the validation, no need for the watcher.